### ISSUE TITLE
Count non-ASCII characters correctly for column numbers

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#63]
quick-lint-js gives incorrect column numbers when the source code contains non-ASCII characters:

```javascript
/*☃*/x
```

Expected:
```
$ ./build/quick-lint-js hello.js
hello.js:1:6: error: use of undeclared variable: x
```

Observed:
```
$ ./build/quick-lint-js hello.js
hello.js:1:8: error: use of undeclared variable: x
```

quick-lint-js' locator should count ☃ as one column, not three.

- [ ] CLI
- [ ] WASM demo
- [x] LSP
- [x] Vim

**Comment by strager** [comment]
* CLI: Count bytes, like GCC and Clang do. Tested in commit 4c2383d2b. Newline handling untested.
* WASM demo: Count UTF-16 code units. Implemented in commit bb83ba4d4.
* LSP: Count UTF-16 code units. Implemented in commit ee99deba9.
* Vim: Count bytes. Tested in commit edcf38eb5. Newline handling poorly tested.
